pkg/controllers/management/auth/globalroles/enqueue.go

```diff
@@ -126,3 +154,82 @@ func (g *globalRBACEnqueuer) crtbEnqueueGRB(_, _ string, obj runtime.Object) ([]
 		{Name: grbOwner},
 	}, nil
 }
+
+// roleEnqueueGR enqueues GlobalRoles that own a given Role when that Role is changed. Uses grOwnerLabel
+// which is protected by the webhook rather than the ownerReference.
+func (g *globalRBACEnqueuer) roleEnqueueGR(_, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
```

Comment:
I'm not familiar enough with wrangler/lasso yet. What happens if the grOwnerLabel is deleted or modified in the `Role` object? Will this enqueuer run for both the old and new object? (old value & new value) Or will nothing be enqueued?

For example, let's say I remove the label and modify the rules by mistake. Do we want Rancher to reconcile the GR again?

Reply:
So as far as I understand it, these enqueue methods will run whenever a Role/RoleBinding get modified/created. It then runs the reconciler on any GR/GRBs that get returned. This all happens after the Role/RoleBinding gets modified, so it would be the new version of the Role/RoleBinding.

In your example where the `grOwnerLabel` gets removed, it wouldn't enqueue the GR. However, in a follow up webhook issue, we are adding a check to prevent users from modifying the `grOwnerLabel` https://github.com/rancher/rancher/issues/42216

Modifying the rules goes through the GR/GRB `Updated` functions, which call `reconcileNamespacedRules` so any changes would update the Roles/RoleBindings

Reply:
I see, the webhook will prevent modifying/deleting the owner label.

> Modifying the rules goes through the GR/GRB

I meant modifying the rule of the `Role`, so in that case that would have to go through the `Role`-to-`GlobalRole` enqueuer first. But since  owner label is always valid, no need to worry about this.

Thanks both of you :+1: